temporian/implementation/numpy/data/event.py

```diff
@@ -25,81 +25,148 @@
     int: DType.INT64,
 }
 
+DTYPE_MAPPING = {
+    np.float64: DType.FLOAT64,
+    np.float32: DType.FLOAT32,
+    np.int64: DType.INT64,
+    np.int32: DType.INT32,
+    np.str_: DType.STRING,
+    np.string_: DType.STRING,
+    np.bool_: DType.BOOLEAN,
+}
+DTYPE_REVERSE_MAPPING = {
+    DType.FLOAT64: np.float64,
+    DType.FLOAT32: np.float32,
+    DType.INT64: np.int64,
+    DType.INT32: np.int32,
+    DType.STRING: np.str_,
+    DType.BOOLEAN: np.bool_,
+}
+
+
+@dataclass
+class IndexData:
+    features: List[np.ndarray]
+    timestamps: np.ndarray
+
+    def __init__(
+        self,
+        features: Union[np.ndarray, List[np.ndarray]],
```

Comment:
I would not allow for the user to feed a single np.array.

Reply:
why not? I think it can make the api less verbose (we parse features to a list in line 57, it won't be stored as a standalone array)

Reply:
This is a nit, but here is my though process.
Keep in mind that this is your code, so you can disagree :).
When I disagree and I want to continue with the code submission, I generally just write "Ack" for acknowledged and close the conversation.

=============
Though process:

This function is not part of the API, so only is will use it. Therefore, code readability and validation is strictly more important than code ease of use. readability >> writing code.

Function with overloading arguments, especially when we start composing them, and especially when we don't have a type checker (yet), can hurt code readability and validation. For this reason, I generally try to avoid overloading arguments when not necessary.

In a way, this is similar to the op definitions. I always advocated for the op constructor not to have overloaded parameters, and instead, to keep overloading arguments strictly for the function API.

Reply:
Very good explanation, I agree. Change's implemented.